File: Topic 01 Patterns/Pattern_21_Hollow_Rectangle_Pattern.cpp
#include <bits/stdc++.h>
using namespace std;

// *****
// *   *
// *   *
// *   *
// *****

void hollowRectanglePattern(int n)
{
    for(int i = 1; i <= n; i++){

        for(int j = 1; j <= n; j++){
            if(i == 1 || j == 1 || i == n || j == n )
                cout << "*";
            else
                cout << " ";
        }

        cout << endl;
    }
}

int main() {
    int n;
    cout << "Enter the number of rows for the Hollow Rectangle Pattern: ";
    cin >> n;

    hollowRectanglePattern(n);

    return 0;
}

File: Java Data/T 01 Patterns/Pattern_21_Hollow_Rectangle_Pattern.java
import java.util.Scanner;

// *****
// *   *
// *   *
// *   *
// *****

public class Pattern_21_Hollow_Rectangle_Pattern {

    public static void hollowRectanglePattern(int n) {

        // Outer loop controls the rows
        for (int i = 1; i <= n; i++) {

            for (int j = 1; j <= n; j++) {

                // Check if the current position is on the border
                if (i == 1 || j == 1 || i == n || j == n) {
                    System.out.print("*");
                } else {
                    System.out.print(" ");
                }
            }

            // Move to the next line
            System.out.println();
        }
    }

    public static void main(String[] args) {

        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter the size of the square (number of rows/columns): ");
        int n = scanner.nextInt();

        hollowRectanglePattern(n);

        scanner.close();

    }
}
